### PR TITLE
Change urllib2 to urllib3

### PR DESCRIPTION
**Pull request by thenets**
I followed the security steps recommend on "urllib3" and "certifi" documentation.
This new version had all security fixes for TLS 1.2 and use "certifi" plugin certificates.

And I also create a script to generate PEM file with custom certificates. You can check at:
https://github.com/thenets/ckanext-harvest/tree/dadosgovbr/ca-custom

I need help to test all those changes.

### CODE REVIEW

**Comment by thenets** [comment]
The test returns: `OK (SKIP=3)`
I used this container for CKAN:
https://github.com/thenets/Easy-CKAN/blob/dev/src/docker/2.6/ckan/Dockerfile

**Comment by wardi** [comment]
@tino097 will take a look

File: pip-requirements.txt [hunk 1]
```diff
@@ -1,2 +1,5 @@
 pika==0.9.8
 redis==2.10.1
+urllib3
+certifi
+requests[security]
```
Comment (tino097) [inline pip-requirements.txt]: overall it is looking good, it will be good to add package version to reduce chance for conflicts with other extensions

Comment (thenets) [inline pip-requirements.txt]: Ok. I'll do it. Can I set the current latest version?

Comment (tino097) [inline pip-requirements.txt]: latest stable i guess will be best

Comment (thenets) [inline pip-requirements.txt]: Ok! I'll commit it today.

**Comment by thenets** [comment]
Python community from IRC recommended me to use the "requests" lib instead of the "urllib3". I'll migrate all modification to this lib and notify you when I finish.

**Comment by amercader** [comment]
+1 just use requests and it will save you a lot of troube.

I personally like to keep support for older versions for as long as possible in extensions. If you feel strongly about it you can remove tests for 2.2 and 2.3, but 2.4 and 2.5 should still be supported.

**Comment by thenets** [comment]
@amercader I'll add support to requests. I learned more about it just last week.
About support, I tried to install and doesn't work. Travis CI got success but I can't install into my VM. I don't know what should I do.

**Comment by amercader** [comment]
@thenets just work on the latest version that you use for general development, and let Travis handle the different versions.

**Comment by thenets** [comment]
@amercader Hm, ok. I'll add those versions again. I'm working on requests now.

**Comment by brucebolt** [comment]
I'm wondering if there is any plan to get this PR included into ckanext-harvest?

**Comment by amercader** [comment]
@brucebolt not as is, but it would be great to migrate to requests. It should be trivial, just replacing [this urllib2](https://github.com/ckan/ckanext-harvest/blob/f715aa5e1271bbc758de0ee72cc636577351a5aa/ckanext/harvest/harvesters/ckanharvester.py#L37) usage with `requests.get`() and adapt the exception codes

**Pull request closed without merging**